stable/21.x: [Coro] RetconOnceDynamic: Distinguish frame allocs and pass frame pointers.

nate-chandler opened the pull request:
To enable allocator implementations that store context in a frame header.

nate-chandler committed:
[Coro] Distinguish frame allocation functions.

Doing so enables implementations to allocate extra space in a header.

nate-chandler committed:
[NFC] Fix variable name.

Deallocation has an address to be deallocated, not a size.

nate-chandler committed:
[Coro] RetconOnceDynamic: Annotate alloc param.

Mark the allocator parameter passed to the allocation functions
swiftcoro.

nate-chandler commented:
https://github.com/swiftlang/swift/pull/85055

@swift-ci test

nate-chandler commented:
https://github.com/swiftlang/swift/pull/85055

@swift-ci test

nate-chandler commented:
@swift-ci test llvm

nate-chandler commented:
@swift-ci test llvm

aschwaighofer reviewed (approved):
Looks good to me

Can you add a test that uses the `llvm.coro.alloca.alloc/free.frame` intrinsics?

nate-chandler committed:
[Coro] RetconOnceDynamic: Pass frame to de/alloc.

Doing so enables allocator implementation to use the frame to store
contextual info.

nate-chandler commented:
Fixed.

nate-chandler commented:
@swift-ci test llvm

nate-chandler commented:
https://github.com/swiftlang/swift/pull/85055

@swift-ci test
